core/Config.php

```diff
@@ -410,13 +410,15 @@ protected function writeConfig()
                 throw $this->getConfigNotWritableException();
             }
 
+            if (!$this->sanityCheck($localPath, $output)) {
+                // If sanity check fails, try to write the contents once more before logging the issue.
+                if (@file_put_contents($localPath, $output, LOCK_EX === false)) {
```

Comment:
@JasonMortonNZ I think there's a typo re the `=== false` which should be outside the function call?

Was also wondering if we should maybe check again the `if` to also do the sanity check again but not 100% sure. It may be useful I suppose but not sure if there could be any issues with that. 

```php
if (@file_put_contents($localPath, $output, LOCK_EX) === false
                    || !$this->sanityCheck($localPath, $output)) {
```